pkg/statistics/handle/autoanalyze/refresher/refresher.go

```diff
@@ -281,10 +320,144 @@ func checkIndexesNeedAnalyze(
 	return indexes
 }
 
+func createTableAnalysisJobForPartitions(
+	sctx sessionctx.Context,
+	tableSchema string,
+	tblInfo *model.TableInfo,
+	tblStats *statistics.Table,
+	defs []model.PartitionDefinition,
+	partitionStats map[int64]*statistics.Table,
+	autoAnalyzeRatio float64,
+	currentTs uint64,
+) *priorityqueue.TableAnalysisJob {
+	// TODO: figure out how to check the table stats version correctly for partitioned tables.
+	tableStatsVer := sctx.GetSessionVars().AnalyzeVersion
+	statistics.CheckAnalyzeVerOnTable(tblStats, &tableStatsVer)
+
+	averageChangePercentage, avgSize, minLastAnalyzeDuration, partitionNames := calculateIndicatorsForPartitions(
+		tblInfo,
+		partitionStats,
+		defs,
+		autoAnalyzeRatio,
+		currentTs,
+	)
+	partitionIndexes := checkIndexesNeedAnalyzeForPartitionedTable(
+		tblInfo,
+		defs,
+		partitionStats,
+	)
+
+	job := &priorityqueue.TableAnalysisJob{
+		TableID:              tblInfo.ID,
+		TableSchema:          tableSchema,
+		TableName:            tblInfo.Name.O,
+		TableStatsVer:        tableStatsVer,
+		ChangePercentage:     averageChangePercentage,
+		TableSize:            avgSize,
+		LastAnalysisDuration: minLastAnalyzeDuration,
+		Partitions:           partitionNames,
+		PartitionIndexes:     partitionIndexes,
+	}
+
+	return job
+}
+
+// calculateIndicatorsForPartitions calculates the average change percentage,
```

Comment:
Please explain the calculation formulas for each parameter in the comments.

Reply:
Added. 
Thanks for your review!  💚 💙 💜 💛 ❤️